WordPress/gutenberg
# Block API: Ensure backwards compatibility for block matchers

with #2854 it's no longer possible to use attribute matchers to define the block attributes. This PR adds a way to ensure backwards compatibility while warning about the "deprecated" status of these matchers.

**Testing instructions**

 - Update any block's attributes definition to the old matchers signature (example: `source.attr( 'myattribute' )`)
 - Load Gutenberg
 - the block should still work as expected
 - A warning should show up in the console when loading the editor.
